source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -780,25 +819,20 @@ namespace
             for (InputPtr input : surfaceShader->getActiveInputs())
             {
                 BindInputPtr bindInput = shaderRef->addBindInput(input->getName(), input->getType());
-                if (input->hasNodeName() && input->hasOutputString())
+                if (input->hasNodeGraphName() && input->hasOutputString() && doc->getNodeGraph(input->getNodeGraphName()))
                 {
-                    if (doc->getNodeGraph(input->getNodeName()))
-                    {
-                        bindInput->setNodeGraphString(input->getNodeName());
-                        bindInput->setOutputString(input->getOutputString());
-                    }
-                    else
-                    {
-                        const auto outputName = std::string(OUTPUT_ELEMENT_PREFIX.c_str()) +
-                                                input->getNodeName() + "_" + 
-                                                input->getOutputString();
-                        if (!doc->getOutput(outputName)) {
-                            auto output = doc->addOutput(outputName, input->getType());
-                            output->setNodeName(input->getNodeName());
-                            output->setOutputString(input->getOutputString());
-                        }
-                        bindInput->setOutputString(outputName);
+                    bindInput->setNodeGraphString(input->getNodeGraphName());
+                    bindInput->setOutputString(input->getOutputString());
+                }
+                else if(input->hasNodeName())
+                {
+                    const auto outputName = std::string(OUTPUT_ELEMENT_PREFIX.c_str()) +
+                                            input->getNodeName() + "_out";
+                    if (!doc->getOutput(outputName)) {
+                        auto output = doc->addOutput(outputName, input->getType());
+                        output->setNodeName(input->getNodeName());
```

Comment:
The newly created output needs a setOutputString if the node has multiple outputs.